spring-core/src/main/java/org/springframework/util/ClassUtils.java

```diff
@@ -138,6 +140,9 @@
 				Object.class, Object[].class, Class.class, Class[].class);
 		registerCommonClasses(Throwable.class, Exception.class, RuntimeException.class,
 				Error.class, StackTraceElement.class, StackTraceElement[].class);
+		registerCommonClasses(Enum.class, Optional.class);
+		registerCommonClasses(Collection.class, List.class, Map.class, Set.class);
```

Comment:
I would go even further here and register Iterable and Iterator as well (collapsed and reordered):

```
registerCommonClasses(Collection.class, List.class, Set.class, Map.class,
        Iterable.class, Iterator.class, Enum.class, Optional.class);
```